lib/xapi-stdext-pervasives/pervasiveext.ml

```diff
@@ -15,14 +15,23 @@
  * Even if fct raises an exception, clean_f is applied
 *)
 
+let src = Logs.Src.create "pervasiveext" ~doc:"logs from Xapi_stdext_pervasives.Pervasiveext"
+
 
 let finally fct clean_f =
   let result =
     try
       fct ();
     with exn ->
       Backtrace.is_important exn;
-      clean_f ();
+      begin
+        (* We catch and log exceptions raised by clean_f to avoid shadowing
+           the original exception raised by fct *)
+        try
+          clean_f ();
+        with cleanup_exn ->
+          Logs.warn ~src (fun m -> m "finally: Error while running cleanup after failure of main function: %s" (Printexc.to_string cleanup_exn));
```

Comment:
Minor: you could use `%a` and `Fmt.exn` instead of `Printexc.to_string`, but its fine either way, we're not using Fmt here yet, so no need to introduce it.